module-system/sov-modules-wallet-blueprint/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+use sov_modules_api::Context;
+use sov_rollup_interface::da::DaSpec;
+
+#[cfg(feature = "cli")]
+pub mod cli;
+
+/// Blueprint definition for a module wallet.
+///
+/// The associated types of this trait are expected to be implemented concretely as binary
+/// endpoints depends on resolved generics in order to compile properly.
+///
+/// The `DefaultWalletBlueprint` contains concrete types for the default implementations on
+/// `sov-modules-api`. It lives behind a feature flag 'default-impl'.
+pub trait WalletBlueprint {
+    /// Context used to define the asymetric cryptography for keys generation and signing.
+    type Context: Context;
+    /// DA specification used to declare runtime call message of the module, that is signed by the
+    /// wallet.
+    type DaSpec: DaSpec;
+}
+
+#[cfg(feature = "default-impl")]
+pub struct DefaultWalletBlueprint;
```

Comment:
We shouldn't need this. If we move the code to `sov-modules-rollup-blueprint` like suggested in another comment we will just implemnt the `WalletBlueprint` for `CelestiaDemoRollup` & `MockDemoRollup` the same way as it was done for `CliWallet`.

Reply:
The goal of this struct is to make a `(Context, DaSpec)` available for crates outside our demos. Meaning: if we create a new runtime that is not the one of celestia or mock, we can use `DefaultWalletBlueprint` to generate the WASM module with `ed25519` keys.

If we implement `WalletBlueprint` for celestia/mock, we will be able to generate WASM only for these runtimes